lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -103,16 +111,12 @@ def status
     end
 
     def type
-      @type || self.class.name.split('::').last
-    end
-
-    def deploy_finished?
-      deploy_failed? || deploy_succeeded? || deploy_timed_out?
+      @type || self.class.name.demodulize
```

Comment:
TIL `demodulize` - Ruby does actually have a function for everything.

Reply:
This is actually coming from ActiveSupport, but you're still not wrong 😉